subprojects/docs/src/docs/userguide/extending-gradle/task_configuration_avoidance.adoc

```diff
@@ -57,6 +57,23 @@ If you are configuring a task by name, you will need to use the configuration av
 
 In the event you still need to get access to a Task instance, you can use link:{javadocPath}/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-[TaskCollection.named(java.lang.String)] and call link:{javadocPath}/org/gradle/api/provider/Provider.html#get--[Provider.get()].  This will cause the task to be created/configured, but everything should work as it has with the eager APIs.
 
+[[sec:how_do_i_order_tasks]]
+=== How do I order tasks with configuration avoidance in mind?
+
+[NOTE]
+====
+Calling ordering methods alone will _not_ cause task creation, as they only declare relationships, but these relationships _might_ cause task creation later in configuration time.
+====
+
+In case, when task relationships need to be established (i.e. `dependsOn`, `finalizedBy`, `mustRunAfter`, `shouldRunAfter`), a distinction can be made between soft, and strong relationships, as their effects on task creation in configuration time differ:
+
+link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:mustRunAfter(java.lang.Object++[]++)[Task.mustRunAfter(...)] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:shouldRunAfter(java.lang.Object++[]++)[Task.shouldRunAfter(...)] represent soft relationships, which only _order_ execution between the related tasks, if the referred tasks exist in the same execution. They will not, in any context, cause task creation in configuration time.
```

Comment:
"... represent soft relationships, which _can_ only change the order of existing tasks, but can't trigger their creation."